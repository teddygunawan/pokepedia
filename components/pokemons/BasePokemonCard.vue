<template functional>
  <div class="card">
    <div class="card-image pokemon-image">
      <figure class="image is-4by3">
        <nuxt-link :to="'/pokemons/' + props.pokemon.id">
          <img :src="props.pokemon.image" alt="PokÃ©mon" />
        </nuxt-link>
      </figure>
    </div>
    <header class="card-header">
      <p class="card-header-title is-capitalized">{{ props.pokemon.name }}<br /></p>
    </header>
    <div class="card-content">
      <div class="content">
        <div class="field has-addons">
          <p
            class="control is-marginless"
            v-for="pokemonType in props.pokemon.types"
            :key="pokemonType.slot"
          >
            <button class="button is-borderless" :class="'type-' + pokemonType.type.name">
              <span>{{ pokemonType.type.name }}</span>
            </button>
          </p>
        </div>
        <!-- <div class="buttons">
          <b-button
            rounded
            :class="'type-' + pokemonType.type.name"
            v-for="pokemonType in pokemon.types"
            :key="pokemonType.slot"
          >
            {{ pokemonType.type.name }}
          </b-button>
        </div> -->
        <!-- Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus nec
        iaculis mauris.
        <a href="#">@bulmaio</a>. <a href="#">#css</a>
        <a href="#">#responsive</a>
        <br />
        <time datetime="2016-1-1">11:09 PM - 1 Jan 2016</time> -->
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    pokemon: { required: true }
  }
}
</script>

<style scoped>
.pokemon-image {
  padding: 15px;
  transition: all 0.5s;
}
.pokemon-image:hover {
  opacity: 0.8;
}
.card-content {
  padding: 0.75rem;
}
</style>
