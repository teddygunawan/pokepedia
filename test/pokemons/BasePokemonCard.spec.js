import { mount, RouterLinkStub } from '@vue/test-utils'
import BasePokemonCard from '@/components/pokemons/BasePokemonCard.vue'

const pokemon = {
  'id': 1,
  'image': 'https://pokeres.bastionbot.org/images/pokemon/1',
  'name': 'bulbasaur',
  'types': []
}

describe('BasePokemonCard', () => {
  test('renders correctly', () => {
    const wrapper = mount(BasePokemonCard, {
      propsData: {
        pokemon: pokemon
      },
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })
    expect(wrapper.find('.pokemon-image').exists()).toBeTruthy()
  })
})
