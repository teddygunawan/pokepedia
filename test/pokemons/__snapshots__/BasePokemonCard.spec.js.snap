// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePokemonCard renders correctly 1`] = `
<div
  class="card"
>
  <div
    class="card-image pokemon-image"
  >
    <figure
      class="image is-4by3"
    >
      <nuxt-link
        to="/pokemons/1"
      >
        <img
          alt="PokÃ©mon"
          src="https://pokeres.bastionbot.org/images/pokemon/1"
        />
      </nuxt-link>
    </figure>
  </div>
   
  <header
    class="card-header has-text-centered"
  >
    <p
      class="card-header-title is-capitalized"
    >
      
      bulbasaur
      <br />
    </p>
  </header>
   
  <div
    class="card-content"
  >
    <div
      class="content"
    >
      <div
        class="field has-addons"
      />
    </div>
  </div>
</div>
`;
